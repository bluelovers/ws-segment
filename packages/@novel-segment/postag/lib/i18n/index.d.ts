export declare const enName: (p: number | "BAD" | "D_A" | "D_B" | "D_C" | "D_D" | "D_E" | "D_F" | "D_I" | "D_L" | "A_M" | "D_MQ" | "D_N" | "D_O" | "D_P" | "A_Q" | "D_R" | "D_S" | "D_T" | "D_U" | "D_V" | "D_W" | "D_X" | "D_Y" | "D_Z" | "A_NR" | "A_NS" | "A_NT" | "A_NX" | "A_NZ" | "D_ZH" | "D_K" | "URL" | "UNK") => string;
export declare const chsName: (p: number | "BAD" | "D_A" | "D_B" | "D_C" | "D_D" | "D_E" | "D_F" | "D_I" | "D_L" | "A_M" | "D_MQ" | "D_N" | "D_O" | "D_P" | "A_Q" | "D_R" | "D_S" | "D_T" | "D_U" | "D_V" | "D_W" | "D_X" | "D_Y" | "D_Z" | "A_NR" | "A_NS" | "A_NT" | "A_NX" | "A_NZ" | "D_ZH" | "D_K" | "URL" | "UNK") => string;
export declare const zhName: (p: number | "BAD" | "D_A" | "D_B" | "D_C" | "D_D" | "D_E" | "D_F" | "D_I" | "D_L" | "A_M" | "D_MQ" | "D_N" | "D_O" | "D_P" | "A_Q" | "D_R" | "D_S" | "D_T" | "D_U" | "D_V" | "D_W" | "D_X" | "D_Y" | "D_Z" | "A_NR" | "A_NS" | "A_NT" | "A_NX" | "A_NZ" | "D_ZH" | "D_K" | "URL" | "UNK") => string;
