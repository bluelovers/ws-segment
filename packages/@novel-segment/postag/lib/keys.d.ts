import POSTAG from './postag/ids';
import { IEnumKeyOf } from './types';
export declare const POSTAG_KEYS: ("BAD" | "D_A" | "D_B" | "D_C" | "D_D" | "D_E" | "D_F" | "D_I" | "D_L" | "A_M" | "D_MQ" | "D_N" | "D_O" | "D_P" | "A_Q" | "D_R" | "D_S" | "D_T" | "D_U" | "D_V" | "D_W" | "D_X" | "D_Y" | "D_Z" | "A_NR" | "A_NS" | "A_NT" | "A_NX" | "A_NZ" | "D_ZH" | "D_K" | "URL" | "UNK")[];
export declare type IPOSTAG_KEYS = IEnumKeyOf<typeof POSTAG>;
export default POSTAG_KEYS;
