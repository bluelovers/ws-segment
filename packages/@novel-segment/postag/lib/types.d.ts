import type { StringKeyOf } from "ts-enum-util/src/types";
export declare type IEnumLike<T extends Record<StringKeyOf<T>, V>, V extends number | string = number | string> = T;
export declare type IEnumKeyOf<T extends Record<StringKeyOf<T>, V>, V extends number | string = number | string> = StringKeyOf<T>;
