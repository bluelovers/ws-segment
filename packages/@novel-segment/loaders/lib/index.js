"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.requireModule = exports.requireDefault = void 0;
const types_1 = require("./types");
function requireDefault(id, subtype) {
    return requireModule(id, subtype).default;
}
exports.requireDefault = requireDefault;
function requireModule(id, subtype) {
    if (id === 'line' && types_1.isUndefined(subtype))
        return require('../line');
    if (id === 'stopword' && types_1.isUndefined(subtype))
        return require('../stopword');
    if (id === 'jieba' && types_1.isUndefined(subtype))
        return require('../jieba');
    if (id === 'opencc' && types_1.isUndefined(subtype))
        return require('../opencc');
    if (id === 'opencc' && subtype === 'scheme')
        return require('../opencc/scheme');
    if (id === 'segment' && types_1.isUndefined(subtype))
        return require('../segment');
    if (id === 'segment' && subtype === 'synonym')
        return require('../segment/synonym');
    throw new Error(`module not defined. id: ${id}, subtype: ${subtype}`);
}
exports.requireModule = requireModule;
//# sourceMappingURL=index.js.map