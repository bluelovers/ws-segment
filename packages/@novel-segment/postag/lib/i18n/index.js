"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.zhName = exports.chsName = exports.enName = void 0;
const keys_1 = __importDefault(require("../keys"));
const ids_1 = __importDefault(require("../postag/ids"));
const chs_1 = __importDefault(require("../postag/chs"));
const cht_1 = __importDefault(require("../postag/cht"));
const en_1 = __importDefault(require("../postag/en"));
const getPOSTagTranslator_1 = __importDefault(require("../util/getPOSTagTranslator"));
keys_1.default.forEach(function (key) {
    let lc = key.toLowerCase();
    // @ts-ignore
    ids_1.default[lc] = ids_1.default[key];
    // @ts-ignore
    chs_1.default[lc] = chs_1.default[key];
    // @ts-ignore
    cht_1.default[lc] = cht_1.default[key];
    // @ts-ignore
    en_1.default[lc] = en_1.default[key];
});
exports.enName = getPOSTagTranslator_1.default(ids_1.default, en_1.default);
exports.chsName = getPOSTagTranslator_1.default(ids_1.default, chs_1.default);
exports.zhName = getPOSTagTranslator_1.default(ids_1.default, cht_1.default);
//# sourceMappingURL=index.js.map