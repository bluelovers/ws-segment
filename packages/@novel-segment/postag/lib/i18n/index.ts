import POSTAG_KEYS from '../keys';
import POSTAG from '../postag/ids';
import CHSNAME from '../postag/chs';
import ZHNAME from '../postag/cht';
import ENNAME from '../postag/en';
import getPOSTagTranslator from '../util/getPOSTagTranslator';

POSTAG_KEYS.forEach(function (key)
{
	let lc = key.toLowerCase();

	// @ts-ignore
	POSTAG[lc] = POSTAG[key];
	// @ts-ignore
	CHSNAME[lc] = CHSNAME[key];
	// @ts-ignore
	ZHNAME[lc] = ZHNAME[key];
	// @ts-ignore
	ENNAME[lc] = ENNAME[key];
});

export const enName = getPOSTagTranslator(POSTAG, ENNAME);
export const chsName = getPOSTagTranslator(POSTAG, CHSNAME);
export const zhName = getPOSTagTranslator(POSTAG, ZHNAME);
