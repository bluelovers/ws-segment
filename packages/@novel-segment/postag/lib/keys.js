"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.POSTAG_KEYS = void 0;
const tslib_1 = require("tslib");
const ids_1 = tslib_1.__importDefault(require("./postag/ids"));
const ts_enum_util_1 = require("ts-enum-util");
exports.POSTAG_KEYS = ts_enum_util_1.$enum(ids_1.default).getKeys();
exports.default = exports.POSTAG_KEYS;
//# sourceMappingURL=keys.js.map