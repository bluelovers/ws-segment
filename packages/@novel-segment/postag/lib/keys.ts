import POSTAG from './postag/ids';
import { $enum } from "ts-enum-util";
import { IEnumKeyOf, IEnumLike } from './types';

export const POSTAG_KEYS = $enum(POSTAG).getKeys();

export type IPOSTAG_KEYS = IEnumKeyOf<typeof POSTAG>;

export default POSTAG_KEYS;
