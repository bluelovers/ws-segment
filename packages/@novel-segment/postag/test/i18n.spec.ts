import { chsName } from '../lib/i18n';

describe(`chsName`, () =>
{

	test(`0x00000008 | 0x00000010`, () =>
	{

		let actual = chsName(0x00000008 | 0x00000010);

		expect(actual).toMatchSnapshot();

	});

})
