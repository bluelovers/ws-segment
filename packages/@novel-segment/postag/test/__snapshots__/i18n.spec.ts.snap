// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chsName 0x00000008 | 0x00000010 1`] = `"外文字符,其他专名"`;
